snuba/datasets/entities/sessions.py

```diff
@@ -79,6 +100,120 @@ def plus_columns(
+
+
+# `errors > 0`
+has_errors = binary_condition(
+    ConditionFunctions.GT, Column(None, None, "errors"), Literal(None, 0)
+)
+# `distinct_id != NIL`
+did_not_nil = neq(distinct_id, lit_nil)
+# `duration != MAX AND status == 1`
+duration_condition = _and(
+    neq(duration, Literal(None, MAX_UINT32)), eq(status, Literal(None, 1))
+)
+# `status IN (2,3,4)`
+terminal_status = in_condition(status, [Literal(None, status) for status in [2, 3, 4]])
+
+# These here are basically the same statements as the matview query
+sessions_raw_translators = TranslationMappers(
+    columns=[
+        ColumnToCurriedFunction(
+            None,
+            "duration_quantiles",
+            FunctionCall(None, "quantilesIf", quantiles),
+            (duration, duration_condition),
+        ),
+        ColumnToFunction(None, "duration_avg", "avgIf", (duration, duration_condition)),
+        ColumnToFunction(
+            None, "sessions", "sumIf", (quantity, eq(seq, Literal(None, 0))),
+        ),
+        ColumnToFunction(
+            None, "sessions_crashed", "sumIf", (quantity, eq(status, Literal(None, 2)))
+        ),
+        ColumnToFunction(
+            None, "sessions_abnormal", "sumIf", (quantity, eq(status, Literal(None, 3)))
+        ),
+        ColumnToFunction(
+            None,
+            "sessions_errored",
+            "plus",
+            (
+                FunctionCall(
+                    None,
+                    "uniqIf",
+                    (session_id, _and(has_errors, neq(session_id, lit_nil))),
+                ),
+                FunctionCall(
+                    None,
+                    "sumIf",
+                    (quantity, _and(terminal_status, eq(session_id, lit_nil))),
+                ),
+            ),
+        ),
+        ColumnToFunction(None, "users", "uniqIf", (distinct_id, did_not_nil)),
+        ColumnToFunction(
+            None,
+            "users_crashed",
+            "uniqIf",
+            (distinct_id, _and(eq(status, Literal(None, 2)), did_not_nil)),
+        ),
+        ColumnToFunction(
+            None,
+            "users_abnormal",
+            "uniqIf",
+            (distinct_id, _and(eq(status, Literal(None, 3)), did_not_nil)),
+        ),
+        ColumnToFunction(
+            None,
+            "users_errored",
+            "uniqIf",
+            (distinct_id, _and(has_errors, did_not_nil)),
+        ),
+    ]
+)
+
+
+class SessionsQueryStorageSelector(QueryStorageSelector):
+    def __init__(self) -> None:
+        self.materialized_storage = get_storage(StorageKey.SESSIONS_HOURLY)
+        self.raw_storage = get_storage(StorageKey.SESSIONS_RAW)
+
+    def select_storage(
+        self, query: Query, request_settings: RequestSettings
+    ) -> StorageAndMappers:
+        # time_range = get_time_range(query, "started")
+        # print(time_range)
+
+        granularity = extract_granularity_from_query(query, "started") or 3600
+        use_materialized_storage = granularity >= 3600
```

Comment:
Shouldn't it be a multiple of 3600 and not only greater than 3600? Technically the contract of extract_granularity_from_query is that it will find the granularity, nobody guarantees that we won't extend to support other granularities that are not multiple of one hour.